src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -179,8 +179,8 @@ func validateRecaptcha(
 ) *util.JSONResponse {
 	if !cfg.Matrix.RecaptchaEnabled {
 		return &util.JSONResponse{
-			Code: http.StatusBadRequest,
-			JSON: jsonerror.BadJSON("Captcha registration is disabled"),
+			Code: http.StatusServiceUnavailable,
```

Comment:
I think StatusBadRequest is correct here?

Reply:
StatusBadRequest would imply there's something wrong with the received JSON which isn't the case here. Service unavailable seems more appropriate.

Reply:
nah, StatusServiceUnavailable is 503. 5xx errors mean "The server is faulty, please try again."

This needs to be a 4xx error.

Reply:
ah yes, server isn't faulty.
I'm guessing 403 or 404 ?

Reply:
I think `StatusBadRequest` might actually be correct here, seeing as it is a client error if it asked for Recaptcha registration. It shouldn't have done so as recaptcha shouldn't have been listed as one of the approved registration flows.